systemtest/src/test/java/io/strimzi/systemtest/MultipleNamespaceST.java

```diff
@@ -0,0 +1,180 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.File;
+import java.util.List;
+
+import static io.strimzi.test.StrimziExtension.REGRESSION;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.not;
+
+@ExtendWith(StrimziExtension.class)
+@Namespace(MultipleNamespaceST.DEFAULT_NAMESPACE)
+@Namespace(value = MultipleNamespaceST.SECOND_NAMESPACE, use = false)
+@ClusterOperator
+class MultipleNamespaceST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(KafkaST.class);
+    static final String DEFAULT_NAMESPACE = "multiple-namespace-test";
+    static final String SECOND_NAMESPACE = "topic-operator-namespace";
+    private static final String TOPIC = "my-topic";
+    private static final String TOPIC_INSTALL_DIR = "../examples/topic/kafka-topic.yaml";
+    private static final int MESSAGE_COUNT = 20;
+
+    private static Resources classResources;
+    private static Resources secondNamespaceResources;
+
+
+    /**
+     * Test the case where the TO is configured to watch a different namespace that it is deployed in
+     */
+    @Test
+    @Tag(REGRESSION)
+    void testWatchingOtherNamespace() throws InterruptedException {
+        List<String> topics = listTopicsUsingPodCLI(CLUSTER_NAME, 0);
+        assertThat(topics, not(hasItems(TOPIC)));
+
+        deployNewTopic();
+        deleteNewTopic();
+    }
+
+    /**
+     * Test the case where messages are sent to topic in different namespace
+     */
+    @Test
+    @Tag(REGRESSION)
+    void testCollectMessagesFromOtherNamespaceTopic() throws InterruptedException {
+        String nameProducerSource = "send-messages-producer-source";
+        String nameConsumerSource = "send-messages-consumer-source";
+
+        deployNewTopic();
+
+        // Create job to send 100 records using Kafka producer for source cluster
+        LOGGER.info("Sending {} messages to topic: {}", MESSAGE_COUNT, TOPIC);
+        waitForJobSuccess(
+                sendRecordsToClusterJob(CLUSTER_NAME, nameProducerSource, TOPIC, MESSAGE_COUNT, null, false)
+        );
+        LOGGER.info("Trying to receive {} messages from topic: {}", MESSAGE_COUNT, TOPIC);
+        // Create job to read 100 records using Kafka producer for source cluster
+        Job jobReadMessagesForTarget = waitForJobSuccess(
+                readMessagesFromClusterJob(CLUSTER_NAME, nameConsumerSource, TOPIC, MESSAGE_COUNT, null, false)
+        );
+
+        checkRecordsForConsumer(MESSAGE_COUNT, jobReadMessagesForTarget);
+        deleteNewTopic();
+    }
+
+    /**
+     * Test the case when Kafka will be deployed in different namespace
+     */
+    @Test
+    @Tag(REGRESSION)
+    void testDeployKafkaToOtherNamespace() throws InterruptedException {
```

Comment:
This is the case where the `Kafka` resource is in a different NS than the CO is running in? Then `testKafkaInDifferentNsThanClusterOperator()` is imho a better name.

Reply:
Correct, I renamed that test.